cfme/tests/services/test_ansible_workflow_servicecatalogs.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-:
+import pytest
+
+from cfme import test_requirements
+from cfme.services.myservice import MyService
+from cfme.services.service_catalogs import ServiceCatalogs
+from cfme.utils import testgen
+from cfme.utils.blockers import GH
+from cfme.utils.log import logger
+
+
+pytestmark = [
+    test_requirements.service,
+    pytest.mark.tier(2),
+    pytest.mark.ignore_stream('upstream')]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.config_managers(metafunc)
+    new_idlist = []
+    new_argvalues = []
+    for i, argvalue_tuple in enumerate(argvalues):
+        args = dict(zip(argnames, argvalue_tuple))
+
+        if not args['config_manager_obj'].yaml_data['provisioning']:
+            continue
+
+        new_idlist.append(idlist[i])
+        new_argvalues.append(argvalues[i])
+
+    testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope='module')
+
+
+@pytest.fixture(scope="module")
+def config_manager(config_manager_obj):
+    """ Fixture that provides a random config manager and sets it up"""
+    if config_manager_obj.type == "Ansible Tower":
+        config_manager_obj.create(validate=True)
+    else:
+        config_manager_obj.create()
+    yield config_manager_obj
+    config_manager_obj.delete()
+
+
+@pytest.fixture(scope="function")
+def catalog_item(appliance, request, config_manager, dialog, catalog):
+    job_type = 'workflow'
+    config_manager_obj = config_manager
+    provider_name = config_manager_obj.yaml_data.get('name')
+    template = config_manager_obj.yaml_data['provisioning_data'][job_type]
+    catalog_item = appliance.collections.catalog_items.create(
+        appliance.collections.catalog_items.ANSIBLE_TOWER,
+        name=dialog.label,
+        description="my catalog",
+        display_in=True,
+        catalog=catalog,
+        dialog=dialog,
+        provider='{} Automation Manager'.format(provider_name),
+        config_template=template)
+    request.addfinalizer(catalog_item.delete)
+    return catalog_item
+
+
+def test_tower_workflow_item(appliance, config_manager, catalog_item, request):
+    """Tests ordering of catalog items for Ansible Workflow templates
+    Metadata:
+        test_flag: provision
+
+    Polarion:
+        assignee: nachandr
+        initialEstimate: 1/4h
+        casecomponent: Services
+        caseimportance: high
+    """
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    service_catalogs.order()
+    logger.info('Waiting for cfme provision request for service %s', catalog_item.name)
+    cells = {'Description': catalog_item.name}
+    order_request = appliance.collections.requests.instantiate(cells=cells, partial_check=True)
+    order_request.wait_for_request(method='ui')
```

Comment:
Since you are specifying `method='ui'` is there any other method to test? E.g. `rest`?

Reply:
Not at the moment